### PR TITLE
Add separation environmnent

### PR DESCRIPTION
**Pull request by ethax-ross**
[Jira-2939](https://dfedigital.atlassian.net.mcas.ms/jira/software/projects/CPDLP/boards/87?selectedIssue=CPDLP-2939)

### Context

We want to add an additional environment for providers to be able to test against the new endpoints added as part of the NPQ separation work.

### Changes proposed in this pull request

- Add new Rails env for separation

Add a new Rails environment for the separation work and enable separation features (apart from migration as there is no corresponding ECF environment to reference yet). Clean up the `webpacker.yml` file.

- Add terraform for separation environment

Adds terraform configuration for separation environment in the production space.

- Deploy to separation environment on merge to main

We want to deploy to the separation environment when merging to `main`, alongside the `sandbox` and `migration` environments.

- Update environment documentation

Add documentation for the separation environment. Clean up docs in general.

- Fix migration if provider does not exist

On a fresh installation the `School-Led Network` provider will not exist and the migrations fail.

- Run seeds on separation environment

- Revert webpacker.yml changes

This isn't right, but its how its currently setup and for some reason asset pre-compilation doesn't work for these environments. One for another day.

### Guidance to review

URL: https://npq-registration-separation-web.teacherservices.cloud/

### CODE REVIEW

File: global_config/separation.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CONFIG=separation
```
Comment (cwrw) [inline global_config/separation.sh:1]: maybe we can name the file `separation_aks` to keep inline with the other naming currently?

File: db/migrate/20221130142333_add_hint_to_lead_providers.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ def up
 
     school_led_network_lead_provider = LeadProvider.find_by(ecf_id: LeadProvider::ALL_PROVIDERS["School-Led Network"])
 
-    school_led_network_lead_provider.update!(
+    school_led_network_lead_provider&.update!(
```
Comment (cwrw) [inline db/migrate/20221130142333_add_hint_to_lead_providers.rb:7]: urgh thanks for this!

**Review by cwrw — APPROVED** [review]
nicely done! 🙌

**Comment by VenkataChalla554** [comment]
[Create terraform environment for NPQ separation](https://trello.com/c/u3polftu/1710-create-terraform-environment-for-npq-separation)

**Pull request merged**